hollow/src/main/java/com/netflix/hollow/api/producer/validation/RecordCountChangeValidator.java

```diff
@@ -0,0 +1,125 @@
+package com.netflix.hollow.api.producer.validation;
+
+import com.netflix.hollow.api.consumer.data.AbstractHollowDataAccessor;
+import com.netflix.hollow.api.producer.HollowProducer;
+import com.netflix.hollow.core.read.engine.HollowReadStateEngine;
+import com.netflix.hollow.core.read.engine.HollowTypeReadState;
+import java.util.logging.Logger;
+
+import static java.util.Objects.requireNonNull;
+
+public class RecordCountChangeValidator implements ValidatorListener {
```

Comment:
Lets add to the javadoc that this only supports types that have a primary key definition

Reply:
Maybe also throw exception to explicitly notify an incorrect use of the validator?

Why does this only support types with primary key definition?

Reply:
The AbstractHollowDataAccessor will throw an exception if the type doesn't have a primary key associated, that works. 

>Why does this only support types with primary key definition?
Without a primary key we don't know if an add+remove was a modification or not (the way to detect modification is to match added and removed record for primary key)